tests/integration-tests/src/test/java/org/apache/activemq/artemis/tests/integration/jms/multiprotocol/JMSMessageExpiryTest.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.activemq.artemis.tests.integration.jms.multiprotocol;
+
+import javax.jms.Connection;
+import javax.jms.Message;
+import javax.jms.MessageConsumer;
+import javax.jms.MessageProducer;
+import javax.jms.Queue;
+import javax.jms.Session;
+import java.lang.invoke.MethodHandles;
+
+import org.apache.activemq.artemis.core.settings.impl.AddressSettings;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.Timeout;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class JMSMessageExpiryTest extends MultiprotocolJMSClientTestSupport {
+
+   protected static final Logger logger = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+   private final long EXPIRY_DELAY = 9999999L;
+   private final long WINDOW = 20L;
+
+   @Test
+   @Timeout(30)
+   public void testCoreMessageExpiryDelay() throws Exception {
+      testExpiry(createCoreConnection(), DelayType.NORMAL, false);
+   }
+
+   @Test
+   @Timeout(30)
+   public void testAmqpMessageExpiryDelay() throws Exception {
+      testExpiry(createConnection(), DelayType.NORMAL, false);
+   }
+
+   @Test
+   @Timeout(30)
+   public void testOpenWireMessageExpiryDelay() throws Exception {
+      testExpiry(createOpenWireConnection(), DelayType.NORMAL, false);
+   }
```

Comment:
Given the differences in handling of 'large' and 'normal' messages I think both need covered in at least 1 test. Also, given there is explicit handling of expiration in the persisting and reload of durable messages, I think a test which restarts the broker before consumption is probably needed for both also.

Reply:
Added.

Reply:
Looks like the additions only cover Large messages and a restart together. Given the runtime and persistence/reload handling of the messages differs significantly, specifically around expiration, we should have a restart test for a 'normal' message as well, and non-restarting test with a 'large' message.

Reply:
In the most recent push I added three new tests using a large message (one for each protocol) and three new tests restarting the broker. The large message tests don't restart the broker and the tests restarting the broker don't use a large message.

Reply:
Yeah, I got it reversed when reading then later commenting and shoudl ahve said 'seperately', so also reverse the resulting points that there are still gaps in the combination testing (as you just outlined). Ones a bus is likely to fit through given the near completely seperate handling of these things for the two message types.

Reply:
I added 3 more tests that use a large message and restart the broker. That should cover all the permutations.